state.go

```diff
@@ -66,11 +80,13 @@ func (a *Autopilot) gatherNextStateInputs(ctx context.Context) (*nextStateInputs
 	// time a state was generated so we know if we have a state old enough where there is
 	// any chance of seeing servers as stable based off that configured threshold.
 	var firstStateTime time.Time
+	var currentState *State
 	a.stateLock.Lock()
-	if a.state != nil {
+	currentState = a.state
+	a.stateLock.Unlock()
```

Comment:
Could this be a read lock/unlock now?

Reply:
I did one better and just replaced all of it with a call to `GetState`. I noticed `GetState` wasn't utilizing read locks so I went ahead and modified that. Then I saw that a couple other similar places would gain the lock (write) just to get a pointer to the current state, so I modified those to call `GetState` as well.